src/Gateway.php

```diff
@@ -409,33 +433,117 @@ public function start( Payment $payment ) {
 
 		// Issuer.
 		if ( Methods::IDEAL === $request->method ) {
-			$request->issuer = $payment->get_meta( 'issuer' );
+			$params['issuer'] = $payment->get_meta( 'issuer' );
 		}
 
-		// Billing email.
-		$billing_email = ( null === $customer ) ? null : $customer->get_email();
+		// Parameters for payments.
+		if ( $request instanceof PaymentRequest ) {
+			if ( \array_key_exists( 'customer_id', $params ) ) {
+				$request->customer_id = $params['customer_id'];
+			}
+
+			if ( \array_key_exists( 'sequence_type', $params ) ) {
+				$request->set_sequence_type( $params['sequence_type'] );
+			}
+
+			if ( \array_key_exists( 'mandate_id', $params ) ) {
+				$request->set_mandate_id( $params['mandate_id'] );
+			}
+
+			if ( \array_key_exists( 'consumer_name', $params ) ) {
+				$request->consumer_name = $params['consumer_name'];
+			}
+
+			if ( \array_key_exists( 'consumer_account', $params ) ) {
+				$request->consumer_account = $params['consumer_account'];
+			}
+
+			if ( \array_key_exists( 'issuer', $params ) ) {
+				$request->issuer = $params['issuer'];
+			}
+
+			// Locale.
+			if ( null !== $customer ) {
+				$request->locale = LocaleHelper::transform( $customer->get_locale() );
+			}
+
+			// Billing email.
+			$billing_email = ( null === $customer ) ? null : $customer->get_email();
 
 			/**
-		 * Filters the Mollie payment billing email used for bank transfer payment instructions.
-		 *
-		 * @since 2.2.0
-		 *
-		 * @param string|null $billing_email Billing email.
-		 * @param Payment     $payment       Payment.
-		 */
-		$billing_email = \apply_filters( 'pronamic_pay_mollie_payment_billing_email', $billing_email, $payment );
+			 * Filters the Mollie payment billing email used for bank transfer payment instructions.
+			 *
+			 * @since 2.2.0
+			 *
+			 * @param string|null $billing_email Billing email.
+			 * @param Payment     $payment       Payment.
+			 */
+			$billing_email = \apply_filters( 'pronamic_pay_mollie_payment_billing_email', $billing_email, $payment );
 
-		$request->set_billing_email( $billing_email );
+			$request->set_billing_email( $billing_email );
 
-		// Due date.
-		if ( ! empty( $this->config->due_date_days ) ) {
-			try {
-				$due_date = new DateTime( sprintf( '+%s days', $this->config->due_date_days ) );
-			} catch ( \Exception $e ) {
-				$due_date = null;
+			// Due date.
+			if ( ! empty( $this->config->due_date_days ) ) {
+				try {
+					$due_date = new DateTime( sprintf( '+%s days', $this->config->due_date_days ) );
+				} catch ( \Exception $e ) {
+					$due_date = null;
+				}
+
+				$request->set_due_date( $due_date );
+			}
+		}
+
+		// Parameters for orders.
+		if ( $request instanceof OrderRequest ) {
```

Comment:
Should we split this op more?

- start() → order or payment?
  - create_order()
  - create_payment()